osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/DeviceRequestMessageProcessor.java

```diff
@@ -113,7 +114,12 @@ public void processMessage(final ObjectMessage message) throws JMSException {
           messageObject, messageMetadata, this.throttlingClientConfig.permitRejectedDelay());
 
     } catch (final Exception exception) {
-      this.sendErrorResponse(messageMetadata, exception, messageObject);
+      if (exception instanceof DirectlyRetryableException) {
```

Comment:
This should be achieved with an extra catch block. But I would not send the message directly to the queue. Please use the mechanism that is in place.

Reply:
Discussion on retry mechanism in core for 'direct' retry is pending. This case of 'direct' retry is therefor solved in DLMS-adapter component (in similar way retry for throttling is handled). Possibly refactoring is needed on conclusion of pending discussion.